github3/repos/branch.py

```diff
@@ -28,5 +29,51 @@ def _update_attributes(self, branch):
         #: Provides the branch's protection status.
         self.protection = branch.get('protection')
 
+        if 'self' in self.links:
+            self._api = self.links['self']
```

Comment:
That certainly would be worthwhile. On a related note I'll also add a PR to remove the hack for fixing the returned value once github fixes that response.